bindings/jupyroot/python/JupyROOT/helpers/utils.py

```diff
@@ -64,38 +72,29 @@
     }}
 
     // Try loading a local version of requirejs and fallback to cdn if not possible.
-    requirejs_load(base_url + 'static/components/requirejs/require.js', requirejs_success(base_url), function(){{
-        requirejs_load('https://cdnjs.cloudflare.com/ajax/libs/require.js/2.2.0/require.min.js', requirejs_success(base_url), function(){{
-            document.getElementById("{jsDivId}").innerHTML = "Failed to load requireJs";
+    script_load(base_url + 'static/scripts/JSRootCore.js', script_success, function(){{
+        script_load('https://root.cern/js/notebook/scripts/JSRootCore.min.js', script_success, function(){{
```

Comment:
Why are we loading an online version of the script here? Doesn't this break the goal of using the local JSRoot files which are in the ROOT installation?

Reply:
It is fallback solution. If `jupyter` is mis-configured, at least default JSROOT drawing will be provided.
But as first choice local version of JSROOT will be tried.

Reply:
I see, I think I would just prefer to fail if the local JS file can't be found, rather than to silently fall back to an online version of it that might not be compatible with the current ROOT installation (which is the reason why you made the change to use the local installation in the first place).

Reply:
Original idea to use local JSROOT come because of offline usage of `Jupyter`. 
And between does not work and mostly work I would prefer second variant.

Reply:
Yes, indeed the offline usage was one motivation, but the compatibility between the C++ libraries and the JSRoot version was another, right? If we go back to using online JSRoot, even as a fallback solution, the online version could have some incompatibility with the ROOT C++ libraries. I think your initial solution of just relying on what is installed locally was better in that sense.

Reply:
I can provide some error output in browser console, that user can get notice that something went wrong.
But why one should disable fallback solution? 
Moreover, one can use particular JSROOT version (like 5.8.1) and update it when necessary, I will add commit now.
This will provide good match between C++ and JS code.
But I prefer to keep code here.

Reply:
The message to warn the user is good.

Hardcoding the JsROOT version to pick online solves the incompatibility problem that I pointed out, but it implies we will need to remember to change that number for every ROOT release (in case it needs to change). Is there any way that, from Python, we can get what is the JsROOT version used by the current ROOT installation, and use that number instead of the hardcoded one?

On the other hand, the fallback solution is only applied in JupyterLab, but not in classic Jupyter, right? Is there any reason for this?

Reply:
> On the other hand, the fallback solution is only applied in JupyterLab, but not in classic Jupyter, right? Is there any reason for this?

Because with `jupyter notebook` and `require.js` such workaround more sophisticated, I have no good idea how to implement it. With `jupyter lab` it was straightforward.

> Is there any way that, from Python, we can get what is the JsROOT version used by the current ROOT installation, and use that number instead of the hardcoded one?

Standalone JSROOT versions are not fully synchronized with ROOT versions.
I introduce special JSROOT branches for ROOT releases like https://github.com/root-project/jsroot/tree/v6-22-00-patches, but did not publish them independently from ROOT. It may will be required in the future, when ROOT7 graphics will be used more actively and will be modified more often.